src/executables/sysrepo-plugind.c

```diff
@@ -567,6 +602,30 @@ main(int argc, char **argv)
     sd_notify(0, "READY=1");
 #endif
 
+    /* update pid file (if daemon) */
+    if (!debug && pidfile) {
+        int pid_len;
+        if (ftruncate(pidfd, 0)) {
+            error_print(0, "Failed to truncate PID file (%s).", strerror(errno));
+            goto cleanup;
+        }
+
+        if (asprintf(&pid, "%d\n", getpid()) < 0) {
+            error_print(0, "Failed to allocate memory for pid (%s).", strerror(errno));
+            goto cleanup;
+        }
+
+        pid_len = (int) strlen(pid);
```

Comment:
- `strlen()` returns `size_t` and `write(2)` accepts a `size_t`, so why is this casting to an `int`?

Reply:
Because of the comparison with the return value of `write` below. I figured the cast there is more justifiable than in the condition expression below. Together with the line(s) above it, it's obvious that the cast is safe, while
```c
if (write(pidfd, pid, pid_len) < (int) pid_len) {
```
would cause more cognitive work I would imagine...